nwapp/index.js

```diff
@@ -31,26 +40,62 @@ process.on('uncaughtException', function (err) {
 
 var win;
 var mainWindow; // this is the chat window (logged in)
-var mainWindowFocused; // Focus tracker. NWK doesn't have a way to query if the window is in focus
+var mainWindowFocused = false; // Focus tracker. NWK doesn't have a way to query if the window is in focus
 var loginView; // log in form
 
+var autoLauncher = new AutoLaunch({
+  name: 'Gitter'
+});
+
+var checkFileExistSync = function(target) {
+  try {
+    return fs.statSync(target);
+  }
+  catch(err) {
+    return false;
+  }
+
+  return false;
+};
+
 (function () {
+  log.info('execPath:', process.execPath);
+  log.info('argv:', argv);
   log.info('version:', pkg.version);
 
-  if (gui.App.argv.length) {
+  var currentInstallPath = argv['current-install-path'];
+  var newUpdaterExecutablePath = argv['new-executable'];
```

Comment:
Gah, autoupdate is hard. Im not the biggest fan of hitting the disk synchronously on startup if we can avoid it.

```javascript
var legacyCurrentInstallPath;
var legacyNewUpdaterExecutablePath;
if (argv.length === 2 && checkFileExistSync(argv._[0]) && checkFileExistSync(argv._[1])) {
  // we have a pre v3 desktop app attempting to update to v3+
  // remove after August 2016
  legacyCurrentInstallPath = argv._[0];
  legacyNewUpdaterExecutablePath = argv._[1];
}

var currentInstallPath = argv['current-install-path'] || legacyCurrentInstallPath;
var newUpdaterExecutablePath = argv['new-executable'] || legacyNewUpdaterExecutablePath;
```